examples/shader/Program.cs

```diff
@@ -13,10 +13,7 @@ protected Effect(string name)
             myName = name;
         }
 
-        public string Name
-        {
-            get { return myName; }
-        }
+        public string Name => myName;
```

Comment:
The major benefit is native support for OSX and Linux (and wherever else .NET Core can be compiled), so Mono would no longer be needed on those platforms. Additionally, a smaller runtime dependency.
Additionally, .NET Core is much more detached from Visual Studio versions. Even on Windows, VS is not needed, only the Core SDK.

.NET Core 1.x ships with C# 6 support minimum (I don't know if .NET Core 2 will up it to 7 or not).

As for operating systems:
Windows 7+
OSX 10.11+
Linux (not really any change as far as I can tell)

Hopefully @Scellow / @Balnian / @Rohansi can comment further.

My thoughts: MS doesn't support (doesn't want to at least) XP and Vista anymore, does SFML still want to support XP and Vista? I don't know/use OSX enough to have an opinion for it.

If you haven't had a chance to read much else, it's looking like waiting for .NET Core 2.0 to be released (and updating the PR) before merging will be best ([summary](https://en.sfml-dev.org/forums/index.php?topic=22150)).